packages/contracts-bedrock/src/Safe/LivenessGuard.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard, GuardManager } from "safe-contracts/base/GuardManager.sol";
+import { ModuleManager } from "safe-contracts/base/ModuleManager.sol";
+import { GetSigners } from "src/Safe/GetSigners.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+contract LivenessGuard is ISemver, BaseGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @notice Emitted when a new set of signers is recorded.
+    /// @param signers An arrary of signer addresses.
+    event SignersRecorded(bytes32 indexed txHash, address[] signers);
```

Comment:
Consider changing to a singular signer and emitting in a loop, i.e. `event SignerRecorded(bytes32 indexed txHash, address indexed signer)`. Then we can index signer addresses and it makes parsing events slightly easier